src/utils.ts

```diff
@@ -173,6 +173,19 @@ export async function executeScript(
   return terminal
 }
 
+export async function stopJavaProcess(processName: string) {
+  // Stop debugger if running
+  if (os.platform() === 'win32') {
+    // Windows stop debugger if already running
+    child_process.execSync('tskill java 2>nul 1>nul || echo "Java not running"')
+  } else {
+    // Linux/Mac stop debugger if already running and make sure script is executable
+    child_process.exec(
+      `kill -9 $(ps -ef | grep '${processName}' | grep 'jar' | awk '{ print $2 }') || return 0`
+    ) // ensure debugger server not running and
+  }
+}
```

Comment:
I'm not a fan of this technique to kill processes.  When we exec the server, we can store the process ID and send the kill signal to that PID.  I could be running some other java process or even some other Ωedit process on a different port and this code will just murder those processes as collateral damage.

Reply:
Yeah this was mostly used for the daffodil-debugger to make sure another instance of it is running, but you are correct with the addition of omega-edit this could possibly trample that omega-edit instance if running. I would be also for removing this. @mbeckerle @stevedlawrence Do you think this would be good to actually remove?

Reply:
If the extension is _managing_ these server processes, what it starts up, it shall also shut down.

If there is _already_ a daffodil-debugger service running _on the same port_ as configured in VS Code, then consider it _unmanaged_ and use it as-is, and don't shut it down when the debugging session is done.  The same applies to the Ωedit server.  This way the extension _cooperates_ with different instances of the servers if they are running and listening to the configured port, otherwise it will use our internal versions as managed processes.

Reply:
Take a look at [spawnd](https://www.npmjs.com/package/spawnd).

Reply:
we want to limit the amount of deps we add. This is done already whenever we start up our own instances however the above code was for ensure a daffodil-debugger was not running without knowing the pid. This was only ran in one place so removing it shouldn't have much of an effect